### PR TITLE
Fix the version scanner working directory

### PR DESCRIPTION
**Pull request by JimSuplizio**
The big problem here is that the version scanner now reads the yml files to figure out what builds in a given sdk/ServiceDirectory but, the yml files were never added to the sparse-checkout. Thankfully, outside of the library that surfaced this issue, there are no issues in the repository. 

I've added code to the script to fail and report an error if it can't find any ci.yml files as well as some additional diagnostics when verbose logging is turned on.

### CODE REVIEW

File: eng/versioning/pom_file_version_scanner.ps1 [hunk 1]
```diff
@@ -57,6 +57,10 @@ $DependencyTypeForError = "$($DependencyTypeCurrent)|$($DependencyTypeDependency
 $UpdateTagFormat = "{x-version-update;<groupId>:<artifactId>;$($DependencyTypeForError)}"
 $UseVerboseLogging = $PSBoundParameters['Debug'] -or $PSBoundParameters['Verbose']
 
+if ($UseVerboseLogging) {
```
Comment (benbp) [inline eng/versioning/pom_file_version_scanner.ps1:60]: Does this get set when we check `use diagnostic logging` in the pipeline and/or would that be a useful thing to support for this case? (if so should be a one line change I can show)

File: eng/versioning/pom_file_version_scanner.ps1 [hunk 2]
```diff
@@ -150,7 +154,14 @@ function Get-ArtifactsList-Per-Service-Directory {
 
     # Get all of the yml files under sdk/
     $ymlFiles = Get-ChildItem -Path $SdkRoot -Recurse -Depth 3 -File -Filter "ci*yml"
+    if ($ymlFiles.Length -eq 0) {
```
Comment (benbp) [inline eng/versioning/pom_file_version_scanner.ps1:157]: TIL I was pretty sure this value would be null, and it is, but apparently `$null.Length` works in powershell...

File: eng/versioning/pom_file_version_scanner.ps1 [hunk 3]
```diff
@@ -175,6 +186,11 @@ function Get-ArtifactsList-Per-Service-Directory {
                 Write-Error "Processing yml file $($ymlFile.FullName)"
                 Write-Error "$ymlPath already contains an Artifact entry for $libFullName"
             }
+            else {
+                if ($UseVerboseLogging) {
+                    Write-Host "    Adding Artifact: $libFullName"
+                }
+            }
```
Comment (benbp) [inline eng/versioning/pom_file_version_scanner.ps1:193]:
```suggestion
            elseif ($UseVerboseLogging) {
                Write-Host "    Adding Artifact: $libFullName"
            }
```

**Comment by JimSuplizio** [comment]
> Some suggestions, LGTM either way.

Yes, I took all of the suggestions.

**Pull request merged**